lte/gateway/c/core/oai/test/amf/test_amf.cpp

```diff
@@ -259,6 +261,27 @@ TEST(test_smf_context_struct, test_smf_context_creation) {
   delete ue_context;
 }
 
+/* Test for registration reject */
+TEST(test_amf_nas5g_pkt_process, test_amf_registration_reject_msg) {
+  NAS5GPktSnapShot nas5g_pkt_snap;
+  // registration reject message
+  RegistrationRejectMsg reg_rej;
+  reg_rej.extended_protocol_discriminator.extended_proto_discriminator = 0x7e;
+  reg_rej.sec_header_type.sec_hdr                                      = 0;
+  reg_rej.spare_half_octet.spare                                       = 0;
+  reg_rej.message_type.msg_type                                        = 0x44;
+  reg_rej.m5gmm_cause.m5gmm_cause                                      = 23;
+
+  bool encode_res = false;
+
+  uint32_t len = 4;
+
+  encode_res = encode_registration_reject_msg(
+      &reg_rej, nas5g_pkt_snap.registration_reject, len);
+
```

Comment:
can you also decode the encoded buffer and assert that the decoded messages matches with reg_rej?

Reply:
added support for decoding encoded buffer and matching with reg_rej.